scripts/eval/eval.py

```diff
@@ -118,6 +127,7 @@ def evaluate_model(
                                None)
 
         if mosaicml_logger is not None:
+            log_analytics_details(mosaicml_logger, model_cfg.model_name)
```

Comment:
Is model_cfg.model_name the best thing to use here @irenedea @dakinggg ? Any suggestions for getting this standardized?

Reply:
Ah sorry, this is confusing but model_name actually is a user-defined name (or "nickname", rather) for the model being evaluated. This allows us to distinguish the models that are being evaluated. See [here](https://github.com/mosaicml/llm-foundry/blob/main/scripts/eval/yamls/hf_eval.yaml#L11) .

The model name we actually want to use is going to be `model.pretrained_model_name_or_path` if `model.name == 'hf_causal_lm'`. What this means is that the user is evaluating a model from huggingface (hence `hf_causal_lm`) loaded by name or path given by `pretrained_model_name_or_path`. For example, this would be for mpt-7b:
```
  model:
    name: hf_causal_lm
    pretrained_model_name_or_path: mosaicml/mpt-7b
    init_device: mixed
    pretrained: true
```

A few more things I think would be useful to log:
- If `model.name` is `hf_causal_lm`, whether `model.pretrained` is true. This tells us if users are loading in the pretrained weights. Otherwise, they may be specifying `load_path` or randomly initializing (like training from scratch)
- Whether `load_path` is provided (see [here](https://github.com/mosaicml/llm-foundry/blob/main/scripts/eval/yamls/mpt_eval.yaml#L27), for example). This tells us if users are loading model weights from a composer checkpoint. 
- Whether `name == 'mpt_causal_lm'`. This would give us signal on whether a user is bringing their own custom mpt-defined architecture.